gufe/protocols/protocolunit.py

```diff
@@ -30,29 +32,11 @@ class Context:
     shared: PathLike
 
 
-class ProtocolUnitMixin:
-
-    def _list_dependencies(self, cls):
-        deps = [] 
-        for key, value in self.inputs.items():
-            if isinstance(value, dict):
-                for k, v in value.items():
-                    if isinstance(v, cls):
-                        deps.append(v)
-            elif isinstance(value, list):
-                for i in value:
-                    if isinstance(i, cls):
-                        deps.append(i)
-            elif isinstance(value, cls):
-                deps.append(value)
-        return deps
-
-
-class ProtocolUnitResultBase(GufeTokenizable, ProtocolUnitMixin):
+class ProtocolUnitResultBase(GufeTokenizable):
     def __init__(self, *, 
             name: Optional[str] = None, 
             source_key: GufeKey, 
-            inputs: Dict[str, Any], 
+            dependencies: Dict[str, GufeKey],
```

Comment:
I think we'll still need a separate `inputs` kwarg here? Otherwise we're dropping some of the `ProtocolUnit` state on the floor when we create its `ProtocolUnitResult`.